mail_tracking/static/src/js/failed_message/discuss.js

```diff
@@ -0,0 +1,397 @@
+                    }
+                    self._mailBus.trigger('update_message', message, data.type);
+                }
+            });
+
+            if (data.needs_actions) {
+                // Increase failed counter if message is marked as failed
+                failed.incrementMailboxCounter(data.message_ids.length);
+            } else {
+                // Decrease failed counter if message is removed from failed
+                failed.decrementMailboxCounter(data.message_ids.length);
+            }
+
+            // Trigger event to refresh threads
+            this._mailBus.trigger('update_failed', failed.getMailboxCounter());
+        },
+    });
+
+    Discuss.include({
+        events: _.extend({}, Discuss.prototype.events, {
+            'click .o_failed_message_retry': '_onRetryFailedMessage',
+            'click .o_failed_message_reviewed': '_onMarkFailedMessageReviewed',
+        }),
+
+        /**
+         * Paramaters used to render 'failed' mailbox entry in Discuss
+         *
+         * @private
+         * @returns {Object}
+         */
+        _sidebarQWebParams: function () {
+            var failed = this.call('mail_service', 'getMailbox', 'failed');
+            return {
+                activeThreadID: this._thread ? this._thread.getID() : undefined,
+                failedCounter: failed.getMailboxCounter(),
+            };
+        },
+
+        /**
+         * Render 'failed' mailbox menu entry in Discuss
+         *
+         * @private
+         * @returns {jQueryElementt}
+         */
+        _renderSidebar: function () {
+            var $sidebar = this._super.apply(this, arguments);
+            // Because Odoo implementation isn't designed to be inherited
+            // properly, we inject 'failed' button using jQuery.
+            var $failed_item = $(QWeb.render('mail_tracking.SidebarFailed',
+                this._sidebarQWebParams()));
+            $failed_item.insertAfter(
+                $sidebar.find(".o_mail_discuss_title_main").filter(":last"));
+            return $sidebar;
+        },
+
+        /**
+         * Overrides to listen click on 'Set all as reviewed' button
+         *
+         * @override
+         */
+        _renderButtons: function () {
+            this._super.apply(this, arguments);
+            this.$btn_set_all_reviewed = this.$buttons.find(
+                '.o_mail_discuss_button_set_all_reviewed');
+            this.$btn_set_all_reviewed
+                .on('click', $.proxy(this, "_onSetAllAsReviewedClicked"));
+        },
+
+        /**
+         * Show or hide 'set all as reviewed' button in discuss mailbox
+         *
+         * This means in which thread the button should be displayed.
+         *
+         * @override
+         */
+        _updateControlPanelButtons: function (thread) {
+            var is_mailbox_failed = thread.getID() === 'mailbox_failed';
+            this.$btn_set_all_reviewed
+                .toggleClass('d-none', !is_mailbox_failed)
```

Comment:
In Bootstrap, `d-md-inline-block` has more importance than `d-none`, but `d-md-none` has more than `d-md-inline-block`.

So, in the Qweb template you can set `d-md-inline-block` statically, and just toggle the `d-none` and `d-md-none` classes:

```suggestion
                .toggleClass('d-none d-md-none', !is_mailbox_failed)
```